lib/spack/spack/solver/asp.py

```diff
@@ -2226,10 +2302,7 @@ def deprecated(self, pkg, version):
     @staticmethod
     def sort_fn(function_tuple):
         name = function_tuple[0]
-        if name == "error":
-            priority = function_tuple[1][0]
-            return (-5, priority)
-        elif name == "hash":
```

Comment:
This needs to be restored for error messages to work properly.

Reply:
This was intentional. See https://github.com/spack/spack/pull/31202/files#diff-50f1229db6f3f9305546f946f508c28f829ab7e778c0a6be4e3a9f4dd6c9b1b7R763-R767